src/components/periodic-report/periodic-report.service.ts

```diff
@@ -101,6 +103,45 @@ export class PeriodicReportService {
         throw new ServerException('Failed to create a periodic report');
       }
 
+      if (
+        input.type === ReportType.Financial ||
+        input.type === ReportType.Narrative
+      ) {
+        await this.db
+          .query()
+          .match(
+            node('project', 'Project', { id: input.projectOrEngagementId })
```

Comment:
These two queries could be combined with a single condition on the label.
```suggestion
            node('node', input.type === ReportType.Progress ? 'Engagement' : 'Project', { id: input.projectOrEngagementId })
```